src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/hooks.ts

```diff
@@ -0,0 +1,110 @@
+import {useFirestoreConfiguration} from '@atb/configuration';
+import {Location} from '@atb/favorites';
+import {CityZone} from '@atb/reference-data/types';
+import {onlyUniquesBasedOnField} from '@atb/utils/only-uniques';
+import turfBooleanPointInPolygon from '@turf/boolean-point-in-polygon';
+import {useMemo} from 'react';
+import {
+  Modes,
+  StreetMode,
+} from '@atb/api/types/generated/journey_planner_v3_types';
+import {useFlexibleTransportEnabled} from './use-flexible-transport-enabled';
+import {StorageModelKeysEnum} from '@atb/storage';
+import {useDebugOverride} from '@atb/debug';
+import {useRemoteConfig} from '@atb/RemoteConfigContext';
+import {RemoteConfigKeys} from '@atb/remote-config';
+
+export const useFindCityZoneInLocation = (
+  location: Location | undefined,
+  cityZones?: CityZone[],
+) => {
+  return useMemo(() => {
+    if (location && cityZones) {
+      const {longitude, latitude} = location.coordinates;
+      return cityZones.find(({geometry}) =>
+        turfBooleanPointInPolygon([longitude, latitude], geometry),
+      );
+    }
+  }, [
+    cityZones,
+    location?.coordinates.longitude,
+    location?.coordinates.latitude,
+  ]);
+};
+
+export const useFindCityZonesInLocations = (
+  from: Location | undefined,
+  to: Location | undefined,
+): CityZone[] | undefined => {
+  const {cityZones} = useFirestoreConfiguration();
+  const fromCityZone = useFindCityZoneInLocation(from, cityZones);
+  const toCityZone = useFindCityZoneInLocation(to, cityZones);
+
+  const filteredCityZones = [fromCityZone, toCityZone].filter(
+    (cityZone) => cityZone && cityZone.enabled === true,
+  ) as CityZone[];
+
+  return filteredCityZones.filter(onlyUniquesBasedOnField('name'));
+};
+
+export function useJourneyModes(
+  defaultValue: StreetMode = StreetMode.Foot,
+): Modes {
+  const flexibleTransportEnabled = useFlexibleTransportEnabled();
+  const flexibleTransportAccessModeEnabled =
+    useFlexibleTransportAccessModeEnabled();
+  const flexibleTransportDirectModeEnabled =
+    useFlexibleTransportDirectModeEnabled();
+  const flexibleTransportEgressModeEnabled =
+    useFlexibleTransportEgressModeEnabled();
```

Comment:
It has been discussed and confirmed we need three different feature toggles for this? We end up adding quite a bit of toggles to Remote Config.

Reply:
Yes it was when we started and now might be that is not needed @tormoseng because this will be always on but in the future. What do you think @tormoseng ?

Reply:
@gorandalum this give us some flexibility for what is being considering for the future, we think is a good to have even now to test, however for future iterations we could consider that is not needed at all, so then we could remove it and will not cause any damage.